quantum/rgb_matrix/post_config.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+// framebuffer
+#if defined(ENABLE_RGB_MATRIX_TYPING_HEATMAP)
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#endif
+#if defined(ENABLE_RGB_MATRIX_DIGITAL_RAIN)
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#endif
+
+// reactive
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SPLASH)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_MULTISPLASH)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_SPLASH)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
+#if defined(ENABLE_RGB_MATRIX_SOLID_MULTISPLASH)
+#    define RGB_MATRIX_KEYPRESSES
+#endif
```

Comment:
instead of doing a train of seperate checks, can't we just do a oneliner? would improve readability
```suggestion
#if (defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS) || \
#    defined(ENABLE_RGB_MATRIX_SPLASH) || \
#    defined(ENABLE_RGB_MATRIX_MULTISPLASH) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_SPLASH) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS) || \
#    defined(ENABLE_RGB_MATRIX_SOLID_MULTISPLASH))
#    define RGB_MATRIX_KEYPRESSES
#endif
```

Reply:
Im not sure its that much more readable, and it was partially that form as generating from <https://github.com/qmk/qmk_firmware/blob/xap/data/constants/lighting/rgb_matrix_0.0.1.hjson> was slightly cleaner.

However ive updated to use a multi-line, as im not pushing the generation right now.